tests/restricted_rooms_test.go

```diff
@@ -77,8 +77,29 @@ func checkRestrictedRoom(t *testing.T, alice *client.CSAPI, bob *client.CSAPI, a
 	})
```

Comment:
The problem in the issue description looks to be related to https://github.com/matrix-org/synapse/issues/15012 (https://github.com/element-hq/synapse/issues/15012)

Cross-linking for those of us who have to deal with this suffering :feelsgood: 

I encountered this in some Complement tests for an internal Element repo (see back-links above for those with Element access)